tests/checkpointing/test_model_checkpoint.py

```diff
@@ -1270,6 +1270,77 @@ def on_load_checkpoint(self, *args, **kwargs):
     make_assertions(cb_restore, written_ckpt)
 
 
+def test_resume_training_preserves_old_ckpt_last(tmpdir):
+    # This test ensures that the last checkpoint when saved is not deleted from the previous folder
+    # (when training is resumed from the old checkpoint)
+
+    from torch.utils.data import DataLoader
+
+    from pytorch_lightning import LightningModule
+
+    class BoringModel(LightningModule):
```

Comment:
You can subclass the BoringModel instead and avoid overriding all other methods thatn `training_step`

```suggestion
    class MyModel(BoringModel):
```